frontends/mit-open/src/pages/OnboardingPage/InterestsStep.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import { Grid, Checkbox, Container } from "ol-components"
+import { InterestsEnum } from "api/v0"
+import CheckBoxOutlined from "@mui/icons-material/CheckBoxOutlined"
+
+import ToggleButtonLabel from "./ToggleButtonLabel"
+import Prompt from "./Prompt"
+import { StepProps } from "./types"
+
+function InterestsStep({ onUpdate, profile }: StepProps) {
+  const [interests, setInterests] = React.useState<InterestsEnum[]>(
+    profile.interests || [],
+  )
+
+  const handleToggle = (event: React.SyntheticEvent, _checked: boolean) => {
+    setInterests((prevInterests) => {
+      const target = event.target as HTMLInputElement
+      if (target.checked) {
+        return [...prevInterests, target.value as InterestsEnum]
+      } else {
+        return prevInterests.filter((interest) => interest !== target.value)
+      }
+    })
+  }
+
+  React.useEffect(() => {
+    onUpdate({ interests })
+  }, [interests, onUpdate])
+
+  return profile ? (
+    <>
+      <h3>
+        Welcome, {profile.name}! What are you interested in learning about?
```

Comment:
I manually created a profile for my superuser. The "name" field on profile appears to be optional. When I loaded the flow without the "name" it looks like this.
<img width="244" alt="Screenshot 2024-05-29 at 3 14 34 PM" src="https://github.com/mitodl/mit-open/assets/196425/d9ffafe2-63f3-441c-b92a-79344894a662"> 

not sure how expected of a state this is but we could conditionalize like so ```Welcome{profile.name? `, ${profile.name}`:''}! What are you interested in learning about?```